crates/runtime-local/src/wasi/extensions.rs

```diff
@@ -21,36 +23,63 @@ use super::guest_error_as_gql;
 pub struct WasiExtensions(Option<Arc<WasiExtensionsInner>>);
 
 impl WasiExtensions {
-    pub fn new(
+    pub async fn new(
         access_log: ChannelLogSender,
         extensions: Vec<ExtensionConfig>,
     ) -> Result<Self, wasi_component_loader::Error> {
         if extensions.is_empty() {
             return Ok(Self(None));
         }
 
-        let mut instance_pools = HashMap::new();
+        let instance_pools = create_pools(access_log, extensions).await?;
+        let inner = WasiExtensionsInner { instance_pools };
+
+        Ok(Self(Some(Arc::new(inner))))
+    }
+}
 
-        for config in extensions {
+async fn create_pools(
+    access_log: ChannelLogSender,
+    extensions: Vec<ExtensionConfig>,
+) -> Result<HashMap<ExtensionId, Pool>, wasi_component_loader::Error> {
+    type Handle = JoinHandle<Result<Option<(ExtensionId, Pool)>, wasi_component_loader::Error>>;
+
+    let mut creating_pools: FuturesUnordered<Handle> = FuturesUnordered::new();
+
+    for config in extensions {
+        let access_log = access_log.clone();
+
+        creating_pools.push(tokio::task::spawn_blocking(move || {
             let manager_config = pool::ComponentManagerConfig {
                 extension_type: config.extension_type,
                 schema_directives: config.schema_directives,
             };
 
             tracing::info!("Loading extension {} {}", config.name, config.version);
 
-            let Some(loader) = ComponentLoader::extensions(config.name, config.wasi_config)? else {
-                continue;
-            };
-
-            let pool = Pool::new(loader, manager_config, config.max_pool_size, access_log.clone());
-            instance_pools.insert(config.id, pool);
-        }
+            match ComponentLoader::extensions(config.name, config.wasi_config)? {
+                Some(loader) => {
+                    let pool = Pool::new(loader, manager_config, config.max_pool_size, access_log);
+                    Ok(Some((config.id, pool)))
+                }
+                None => Ok(None),
+            }
+        }));
+    }
 
-        let inner = WasiExtensionsInner { instance_pools };
+    let mut pools = HashMap::new();
 
-        Ok(Self(Some(Arc::new(inner))))
+    while let Some(result) = creating_pools.next().await {
```

Comment:
Would create a stream and use https://docs.rs/futures-util/latest/futures_util/stream/trait.StreamExt.html#method.buffer_unordered with limit = https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html
This will execute everything in parallel.

Reply:
so 100 extension -> 100 extensions being built all at once today